hw6/hw6/ViewController.swift

```diff
@@ -0,0 +1,142 @@
+        
+        // 4) * Создайте функцию, которая проверяет является ли переданная строка палиндромом. Строка может быть как одним словом (mom) так и предложением (Too bad I hid a boot). В предложении вы должны игнорировать пробелы, большие или маленькие это буквы и знаки препинания при определении полиндрома.
+        
+        print(checkPolyndrom("Too bad I hid a boot!"))
+        print(checkPolyndrom("Too, bad, I, hid, a,-?: boot!"))
+   }
+
+    func add(_ firstNumber: Int?, to secondNumber: Int?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(firstNumber + secondNumber)
+    }
+    
+    func multiply(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        if let firstNumber = firstNumber,
+           let secondNumber = secondNumber {
+            return String(firstNumber * secondNumber)
+        }
+        return "Одно из значений не число"
+    }
+    
+    func myPow(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(pow(Double(firstNumber), Double(secondNumber)))
+    }
+    
+    func subtract(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        String((firstNumber ?? 0) - (secondNumber ?? 0))
+    }
+    
+    func devide(_ firstNumber: Double?, _ secondNumber: Double?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(firstNumber / secondNumber)
+    }
+    
+    func createCharacterArray(_ stringArray: [String?]) -> [Character] {
+        var characterArray = [Character]()
+
+        for element in stringArray {
+            guard let element = element else {
+                return [Character]()
+            }
+
+            for char in element {
+                characterArray.append(char)
+            }
+        }
+        return characterArray
+    }
+    
+    func countNumberRepeatedLettersOptionalStringArray(_ optionalStringArray: [String?]) {
+        let characterArray = createCharacterArray(optionalStringArray)
+        
+        let result = characterArray.reduce(into: [Character: Int]()) { partialResult, element in
+            partialResult[element, default: 0] += 1}
+        
+        print(result)
+    }
+    
+    func checkPolyndrom(_ polyndromString: String?) -> String {
+        guard let polyndromString = polyndromString else {
+            return "Введите строку"
+        }
+        let charactersToRemove = Set(Array(".:?,-!"))
+        let lowerWithoutWhitespacesString = polyndromString.filter{ !$0.isWhitespace }.lowercased().reduce("") { let str = String($1); return $0 + (charactersToRemove.contains($1) ? "" : str) }
```

Comment:
I believe you can optimise this string like this: `polyndromString.filter{ !charactersToRemove.contains($0) }.lowercased()`